notesystem/modes/search_mode.py

```diff
@@ -49,8 +50,13 @@ def _run(self, args) -> None:  # TODO: Should return exit code
             {FileNotFoundError} -- When the given path cannot be found
         """
 
+        if 'tag_delimiter' in args:
+            self.tag_delimiter = args['tag_delimiter']
+        else:
+            self.tag_delimiter = ' '  # Default to space
```

Comment:
Note: tag delimter has a default set but this check is still required because some tests do not run the cmdline but instead create the arguments themselfs